src/Tasks/Microsoft.NET.Build.Tasks/EmbedAppNameInHost.cs

```diff
@@ -59,7 +60,24 @@ protected override void ExecuteCore()
             {
                 Directory.CreateDirectory(destinationDirectory);
             }
-            File.WriteAllBytes(ModifiedAppHostPath, array);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
My suggestion would be to keep the code consistent - there is no overhead scenario in this operation.